src/components/shapes/attributes.js

```diff
@@ -224,6 +225,52 @@ module.exports = templatedArray('shape', {
             '`config.editable` or `config.edits.shapePosition`.'
         ].join(' ')
     },
-
+    label: {
+        text: {
+            valType: 'string',
+            dflt: '',
+            editType: 'arraydraw',
+            description: 'Sets the text to display with shape.'
+        },
+        font: fontAttrs({
+            editType: 'calc+arraydraw',
+            colorEditType: 'arraydraw',
+            description: 'Sets the shape label text font.'
+        }),
+        textposition: {
+            valType: 'enumerated',
+            values: [
+                'top left', 'top center', 'top right',
+                'middle left', 'middle center', 'middle right',
+                'bottom left', 'bottom center', 'bottom right',
+                'start', 'middle', 'end',
+            ],
+            editType: 'arraydraw',
+            description: 'Sets the position of the label text relative to the shape.'
+        },
+        textangle: {
+            valType: 'angle',
+            editType: 'calc+arraydraw',
+            description: [
+                'Sets the angle at which the label text is drawn',
+                'with respect to the horizontal. For lines, the default textangle is `auto`,',
+                'which displays the text at the same angle as the line. For all other shapes,',
+                'the default textangle is 0 (horizontal).'
+            ].join(' ')
+        },
+        xanchor: extendFlat({}, annAttrs.xanchor, {
+            description: 'Sets the text box\'s horizontal position anchor.',
+        }),
+        yanchor: extendFlat({}, annAttrs.yanchor, {
+            description: 'Sets the text box\'s vertical position anchor.',
+        }),
+        padding: {
+            valType: 'number',
+            dflt: 3,
+            editType: 'arraydraw',
+            description: 'Sets padding between edge of label and xanchor / yanchor.'
+        },
+        editType: 'arraydraw'
+    },
```

Comment:
@emilykl Could you possibly add similar `label` options to `newshape`: 
https://github.com/plotly/plotly.js/blob/1625db4299a6798f44490901eb57656cb56bbbcd/src/components/shapes/draw_newshape/attributes.js#L7